hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -90,6 +101,84 @@ public void visitAssignment(Assignment assignment) {
 		return affectedTableNames;
 	}
 
+	@Override
+	public void render(SqlAstNode sqlAstNode, SqlAstNodeRenderingMode renderingMode) {
+		switch ( renderingMode ) {
+			case NO_PLAIN_PARAMETER:
+				if ( sqlAstNode instanceof SqmParameterInterpretation ) {
+					sqlAstNode = ( (SqmParameterInterpretation) sqlAstNode ).getResolvedExpression();
+				}
+				if ( sqlAstNode instanceof JdbcParameter ) {
+					final JdbcMapping jdbcMapping = ( (JdbcParameter) sqlAstNode ).getExpressionType().getJdbcMappings()
+							.get( 0 );
+					// We try to avoid inlining parameters if possible which can be done by wrapping the parameter
+					// in an expression that is semantically unnecessary e.g. numeric + 0 or concat with an empty string
+					switch ( jdbcMapping.getSqlTypeDescriptor().getJdbcTypeCode() ) {
```

Comment:
Do we _really_ not have some utility function for determining if a SQL type code is numeric anywhere?

Reply:
(Or a string.)

Reply:
We don't, but I can introduce it if you want.

Reply:
I would. Makes the method shorter and more readable.